service/types.go

```diff
@@ -106,30 +123,45 @@ func (o *options) UnmarshalJSON(data []byte) error {
 		}
 		o.Nonce = nonce
 	}
+	if len(ow.ContractData) > 0 {
+		o.ContractData = ow.ContractData
+	}
 
 	return nil
 }
 
 type metadata struct {
```

Comment:
I think we should break this struct up like:
```
type metadata struct {
	Nonce           uint64      `json:"nonce"`
	GasPrice        *big.Int    `json:"gas_price"`
	GasLimit        uint64      `json:"gas_limit"`

	UnwrapBridgeTx  bool        `json:"bridge_unwrap"`

	ContractData    string      `json:"data,omitempty"`
	MethodSignature string      `json:"method_signature,omitempty"`
	MethodArgs      interface{} `json:"method_args,omitempty"`
}
```

Reply:
(same with the others)

Reply:
sure!